src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp

```diff
@@ -62,7 +62,6 @@ MergeTreeBackgroundExecutor<Queue>::MergeTreeBackgroundExecutor(
     CurrentMetrics::Metric metric_,
     CurrentMetrics::Metric max_tasks_metric_,
     std::string_view policy)
-    requires requires(Queue queue) { queue.updatePolicy(policy); } // Because we use explicit template instantiation
```

Comment:
```
src/CMakeFiles/dbms.dir/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp.o -c /src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp
Step #3 - "compile-libfuzzer-address-x86_64": [1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:68:13: [0m[0;1;31merror: [0m[1mno member named 'updatePolicy' in 'DB::RoundRobinRuntimeQueue'[0m
Step #3 - "compile-libfuzzer-address-x86_64":     pending.updatePolicy(policy);
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m    ~~~~~~~ ^
Step #3 - "compile-libfuzzer-address-x86_64": [0m[1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:326:16: [0m[0;1;30mnote: [0min instantiation of member function 'DB::MergeTreeBackgroundExecutor<DB::RoundRobinRuntimeQueue>::MergeTreeBackgroundExecutor' requested here[0m
Step #3 - "compile-libfuzzer-address-x86_64": template class MergeTreeBackgroundExecutor<RoundRobinRuntimeQueue>;
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m               ^
Step #3 - "compile-libfuzzer-address-x86_64": [0mIn file included from /src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:1:
Step #3 - "compile-libfuzzer-address-x86_64": [1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.h:277:17: [0m[0;1;31merror: [0m[1mno member named 'updatePolicy' in 'DB::RoundRobinRuntimeQueue'[0m
Step #3 - "compile-libfuzzer-address-x86_64":         pending.updatePolicy(new_policy);
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m        ~~~~~~~ ^
Step #3 - "compile-libfuzzer-address-x86_64": [0m[1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:326:16: [0m[0;1;30mnote: [0min instantiation of member function 'DB::MergeTreeBackgroundExecutor<DB::RoundRobinRuntimeQueue>::updateSchedulingPolicy' requested here[0m
Step #3 - "compile-libfuzzer-address-x86_64": template class MergeTreeBackgroundExecutor<RoundRobinRuntimeQueue>;
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m               ^
Step #3 - "compile-libfuzzer-address-x86_64": [0m[1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:68:13: [0m[0;1;31merror: [0m[1mno member named 'updatePolicy' in 'DB::PriorityRuntimeQueue'[0m
Step #3 - "compile-libfuzzer-address-x86_64":     pending.updatePolicy(policy);
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m    ~~~~~~~ ^
Step #3 - "compile-libfuzzer-address-x86_64": [0m[1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:327:16: [0m[0;1;30mnote: [0min instantiation of member function 'DB::MergeTreeBackgroundExecutor<DB::PriorityRuntimeQueue>::MergeTreeBackgroundExecutor' requested here[0m
Step #3 - "compile-libfuzzer-address-x86_64": template class MergeTreeBackgroundExecutor<PriorityRuntimeQueue>;
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m               ^
Step #3 - "compile-libfuzzer-address-x86_64": [0mIn file included from /src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:1:
Step #3 - "compile-libfuzzer-address-x86_64": [1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.h:277:17: [0m[0;1;31merror: [0m[1mno member named 'updatePolicy' in 'DB::PriorityRuntimeQueue'[0m
Step #3 - "compile-libfuzzer-address-x86_64":         pending.updatePolicy(new_policy);
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m        ~~~~~~~ ^
Step #3 - "compile-libfuzzer-address-x86_64": [0m[1m/src/ClickHouse/src/Storages/MergeTree/MergeTreeBackgroundExecutor.cpp:327:16: [0m[0;1;30mnote: [0min instantiation of member function 'DB::MergeTreeBackgroundExecutor<DB::PriorityRuntimeQueue>::updateSchedulingPolicy' requested here[0m
Step #3 - "compile-libfuzzer-address-x86_64": template class MergeTreeBackgroundExecutor<PriorityRuntimeQueue>;
Step #3 - "compile-libfuzzer-address-x86_64": [0;1;32m               ^
Step #3 - "compile-libfuzzer-address-x86_64": [0m4 errors generated.
Step #3 - "compile-libfuzzer-address-x86_64": 
```

I failed to understand why it doesn't work.
the best I came up with is to provide stubs for this method that throw `NOT_IMPLEMENTED`.

Reply:
yes, we got the error above building fuzzer for cluster fuzz. it uses clang 15.0.0 which is almost one year old. maybe it is the reason, but the fact is that this "sfinae" didn't work, so I provided this method for other classes

Reply:
Okay, let's simplify indeed.